common/src/commands.rs

```diff
@@ -277,63 +344,573 @@ pub struct Stop {}
-            proto_update_wl_state
+            ankaios::Request::try_from(proto_request_complete_state).unwrap(),
+            ankaios_request_complete_state
         );
     }
 
     #[test]
-    fn utest_converts_to_proto_request_complete_state() {
-        let ankaios_request_complete_state = Request {
-            request_id: "42".to_string(),
-            request_content: RequestContent::CompleteStateRequest(CompleteStateRequest {
-                field_mask: vec!["1".to_string(), "2".to_string()],
-            }),
+    fn utest_converts_from_proto_update_state_request_inner_state_with_empty_states() {
+        let mut proto_request_complete_state = update_state_request!(proto);
+        let mut ankaios_request_complete_state = update_state_request!(ankaios);
+
+        let proto::RequestContent::UpdateStateRequest(proto_request_content) =
+            proto_request_complete_state
+                .request_content
+                .as_mut()
+                .unwrap()
+        else {
+            unreachable!()
         };
+        proto_request_content
+            .new_state
+            .as_mut()
+            .unwrap()
+            .startup_state = None;
+        proto_request_content
+            .new_state
+            .as_mut()
+            .unwrap()
+            .current_state = None;
+
+        let ankaios::RequestContent::UpdateStateRequest(ankaios_request_content) =
+            &mut ankaios_request_complete_state.request_content
+        else {
+            unreachable!()
+        };
+        ankaios_request_content.state.startup_state = Default::default();
+        ankaios_request_content.state.current_state = Default::default();
 
-        let proto_request_complete_state = proto::Request {
-            request_id: "42".to_string(),
-            request_content: Some(proto::request::RequestContent::CompleteStateRequest(
-                proto::CompleteStateRequest {
-                    field_mask: vec!["1".to_string(), "2".to_string()],
+        assert_eq!(
+            ankaios::Request::try_from(proto_request_complete_state).unwrap(),
+            ankaios_request_complete_state
+        );
+    }
+
+    #[test]
+    fn utest_converts_from_proto_update_state_request_fails_invalid_startup_state() {
+        let mut proto_request_complete_state = update_state_request!(proto);
+
+        let proto::RequestContent::UpdateStateRequest(proto_request_content) =
+            proto_request_complete_state
+                .request_content
+                .as_mut()
+                .unwrap()
+        else {
+            unreachable!()
+        };
+        proto_request_content
+            .new_state
+            .as_mut()
+            .unwrap()
+            .startup_state
+            .as_mut()
+            .unwrap()
+            .workloads
+            .insert(
+                WORKLOAD_NAME_1.into(),
+                proto::Workload {
+                    update_strategy: -1,
```

Comment:
Could you use different item than `update_strategy` here? The `update_strategy` has been removed in another pull request #171 

The same problem is in the next test too.

Reply:
I just took a look at #171, and it seems after this PR we can implement `From<proto:Request>` instead of `TryFrom<proto:Requeset>` for `Request`, which make this whole test useless. I think it is better to first merge #171, and then adapt the tests here.

Reply:
Ok, I was wrong. We still need to implement `TryFrom<proto:Requeset>`, as the `dependencies` added to `Workload` use `AddCondition`, which is an enum. I now use `AddCondition` to test a failed conversion.